cpp/src/quantiles/tdigest/tdigest_aggregation.cu

```diff
@@ -293,47 +345,79 @@ struct centroid_group_info {
   GroupOffsetsIter group_offsets;
   size_type const* tdigest_offsets;
 
-  __device__ thrust::tuple<double, size_type, size_type> operator()(size_type group_index) const
+  centroid_group_info(double const* _cumulative_weights,
```

Comment:
Same thoughts about this constructor: https://github.com/rapidsai/cudf/pull/19140/files#r2170839063